app/lib/feed_manager.rb

```diff
@@ -246,10 +246,14 @@ def phrase_filtered?(status, receiver_id, context)
     combined_regex = active_filters.reduce { |memo, obj| Regexp.union(memo, obj) }
     status         = status.reblog if status.reblog?
 
-    !combined_regex.match(Formatter.instance.plaintext(status)).nil? ||
-      (status.spoiler_text.present? && !combined_regex.match(status.spoiler_text).nil?) ||
-      (status.preloadable_poll && !combined_regex.match(status.preloadable_poll.options.join("\n\n")).nil?) ||
-      !combined_regex.match(status.media_attachments.map(&:description).join("\n\n")).nil?
+    combined_text = [
+        Formatter.instance.plaintext(status),
+        status.spoiler_text,
+        status.preloadable_poll ? status.preloadable_poll.options.join("\n\n") : nil,
+        status.media_attachments.map(&:description).join("\n\n")
+    ].filter.join("\n\n")
```

Comment:
I think by `filter` you mean `compact`

Reply:
I took the liberty to fix the remaining issues with this PR.